src/routes/write-exercise/components/WriteExercise.tsx

```diff
@@ -41,14 +42,23 @@ const WriteExercise = ({ documents, route, navigation }: WriteExercisePropType):
     }))
   )
 
+  const keyboard = useKeyboard()
   const current = documentsWithResults[currentIndex]
   const nthRetry = current.numberOfTries ?? 0
   const needsToBeRepeated = nthRetry < numberOfMaxRetries && current.result !== SIMPLE_RESULTS.correct
 
   const tryLater = useCallback(() => {
-    setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
-    Keyboard.dismiss()
-  }, [documentsWithResults, currentIndex])
+    // Wait until keyboard is hidden before go next
+    if (keyboard.keyboardShown) {
+      const onKeyboardHideSubscription = Keyboard.addListener('keyboardDidHide', () => {
+        setDocumentsWithResults(moveToEnd(documentsWithResults, currentIndex))
+        onKeyboardHideSubscription.remove()
+      })
+      Keyboard.dismiss()
```

Comment:
Why do you use both the hook and the import from react-native? Do we need the additional dependency?

Reply:
should i add the regular lifecycle function or create my own hook? sth like this?
```js
const [isKeyboardVisible, setKeyboardVisible] = useState(false);

 useEffect(() => {
    const keyboardDidShowListener = Keyboard.addListener(
      'keyboardDidShow',
      () => {
        setKeyboardVisible(true); // or some other action
      }
    );
    const keyboardDidHideListener = Keyboard.addListener(
      'keyboardDidHide',
      () => {
        setKeyboardVisible(false); // or some other action
      }
    );

    return () => {
      keyboardDidHideListener.remove();
      keyboardDidShowListener.remove();
    };
  }, []);
```